File: app/src/main/java/com/sleticalboy/doup/util/ActivityController.java
package com.sleticalboy.doup.util;

import android.app.Activity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Android Studio.
 * Date: 12/27/17.
 *
 * @author sleticalboy
 */

public class ActivityController {

    private static List<Activity> sActivityList = new ArrayList<>();

    public static void add(Activity a) {
        if (!sActivityList.contains(a))
            sActivityList.add(a);
    }

    public static void remove(Activity a) {
        if (sActivityList.contains(a))
            sActivityList.remove(a);
    }

    public static void finishAll() {
        for (Activity a : sActivityList) {
            if (!a.isFinishing())
                a.finish();
        }
        sActivityList.clear();
    }
}

File: my-utils/src/main/java/com/sleticalboy/util/ActivityController.kt
package com.sleticalboy.util

import android.app.Activity
import java.util.*

/**
 * Created by Android Studio.
 * Date: 12/27/17.
 *
 * @author sleticalboy
 */
object ActivityController {

    private val activityList = ArrayList<Activity>()

    @JvmStatic
    fun add(activity: Activity) {
        if (!activityList.contains(activity))
            activityList.add(activity)
    }

    @JvmStatic
    fun remove(activity: Activity) {
        if (activityList.contains(activity))
            activityList.remove(activity)
    }

    @JvmStatic
    fun finishAll() {
        for (activity in activityList) {
            if (!activity.isFinishing) {
                activity.finish()
            }
        }
        activityList.clear()
    }
}
